custom/icds_reports/ucr/expressions.py

```diff
@@ -288,10 +290,31 @@ class GetAppVersion(JsonObject):
     def configure(self, app_version_string):
         self._app_version_string = app_version_string
 
-    def __call__(self, item, context=None):
+    def get_version_from_app_object(self, item, app_version_from_app_string):
+        domain_name = item.get('domain')
+        form_accessor = FormAccessors(domain_name)
+        form_id = item.get('form').get('meta').get('instanceID')
+        form = form_accessor.get_form(form_id)
+
+        app_build = get_build_by_version(domain_name, form.app_id, app_version_from_app_string)
+
+        if app_build is None:
+            return app_version_from_app_string
+
+        else:
```

Comment:
You still need to check if it's a linked app, `upstream_app` will be null for regular apps.

Since it's just the json from saved_build, not an app, just check that `doc_type` is `LinkedApplication`.

Reply:
Good Catch. Thanks